l10n_ec_delivery_note/models/account_edi_format.py

```diff
@@ -0,0 +1,73 @@
+from odoo import _, models
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = "account.edi.format"
+
+    def l10n_ec_is_required_for_delivery_note(self, note):
+        if (
+            note.country_code != "EC"
+            or note.journal_id.l10n_ec_emission_type != "electronic"
+        ):
+            return False
+        if (
+            self.code == "l10n_ec_format_sri"
+            and note.l10n_latam_internal_type.internal_type == "delivery_note"
```

Comment:
This can be a little mess to understand, maybe better use edi_oca and edi_oca_stock implementation, to avoid use misunderstand concepts, Delivery Note is not a financial record, is like a EDI process external to account module, but well it's only my opinion